adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -294,11 +305,22 @@ public void acquireToken(Activity activity, String resource, String clientId,
 
         redirectUri = checkInputParameters(resource, clientId, redirectUri, callback);
 
+        final APIEvent apiEvent = new APIEvent(EventStrings.API_EVENT, mContext, clientId);
+        final String requestId = Telemetry.registerNewRequest();
+        apiEvent.setRequestId(requestId);
+        apiEvent.setAPIId(EventStrings.ACQUIRE_TOKEN_2);
+        apiEvent.setAuthority(getAuthority());
+
+        Telemetry.getInstance().startEvent(requestId, apiEvent.getEventName());
```

Comment:
:) please wrap it into a seperate method, and call the method in each acquire token API call.